src/core/qgsquick/qgsquickcoordinatetransformer.cpp

```diff
@@ -242,52 +253,54 @@ void QgsQuickCoordinateTransformer::setVerticalGrid( const QString &grid )
-    if ( !dataDirs.isEmpty() )
+    if ( QFile::exists( mVerticalGrid ) )
     {
-      for ( const QString &dataDir : dataDirs )
+      mVerticalGridPath = mVerticalGrid;
+    }
+    else
+    {
+      QStringList dataDirs = PlatformUtilities::instance()->appDataDirs();
+      if ( !dataDirs.isEmpty() )
       {
-        const QString verticalGridPath = QStringLiteral( "%1proj/%2" ).arg( dataDir, mVerticalGrid );
-        if ( QFile::exists( verticalGridPath ) )
+        for ( const QString &dataDir : dataDirs )
         {
-          mVerticalGridPath = verticalGridPath;
-          break;
+          const QString verticalGridPath = QStringLiteral( "%1proj/%2" ).arg( dataDir, mVerticalGrid );
+          if ( QFile::exists( verticalGridPath ) )
+          {
+            mVerticalGridPath = verticalGridPath;
+            break;
+          }
         }
       }
     }
-  }
 
-  if ( !mVerticalGridPath.isEmpty() )
-  {
-    GDALDatasetH hDataset;
-    GDALAllRegister();
-    hDataset = GDALOpen( mVerticalGridPath.toUtf8().constData(), GA_ReadOnly );
-    if ( hDataset != NULL )
+    if ( !mVerticalGridPath.isEmpty() )
     {
-      OGRSpatialReferenceH spatialRef = GDALGetSpatialRef( hDataset );
-      char *pszWkt = nullptr;
-      const QByteArray multiLineOption = QStringLiteral( "MULTILINE=NO" ).toLocal8Bit();
-      const QByteArray formatOption = QStringLiteral( "FORMAT=WKT2" ).toLocal8Bit();
-      const char *const options[] = { multiLineOption.constData(), formatOption.constData(), nullptr };
-      OSRExportToWktEx( spatialRef, &pszWkt, options );
-      mCoordinateVerticalGridTransform.setDestinationCrs( QgsCoordinateReferenceSystem::fromWkt( QString( pszWkt ) ) );
-      mCoordinateVerticalGridTransform.setContext( mCoordinateTransform.context() );
-      CPLFree( pszWkt );
-    }
-    else
-    {
-      // Invalid grid file, skip
-      mVerticalGridPath.clear();
+      GDALDatasetH hDataset;
+      GDALAllRegister();
+      hDataset = GDALOpen( mVerticalGridPath.toUtf8().constData(), GA_ReadOnly );
+      if ( hDataset != NULL )
+      {
+        OGRSpatialReferenceH spatialRef = GDALGetSpatialRef( hDataset );
+        char *pszWkt = nullptr;
+        const QByteArray multiLineOption = QStringLiteral( "MULTILINE=NO" ).toLocal8Bit();
+        const QByteArray formatOption = QStringLiteral( "FORMAT=WKT2" ).toLocal8Bit();
+        const char *const options[] = { multiLineOption.constData(), formatOption.constData(), nullptr };
+        OSRExportToWktEx( spatialRef, &pszWkt, options );
+        mCoordinateVerticalGridTransform.setDestinationCrs( QgsCoordinateReferenceSystem::fromWkt( QString( pszWkt ) ) );
+        mCoordinateVerticalGridTransform.setContext( mCoordinateTransform.context() );
+        CPLFree( pszWkt );
+      }
+      else
+      {
+        // Invalid grid file, skip
+        mVerticalGridPath.clear();
+      }
+      GDALClose( hDataset );
     }
-    GDALClose( hDataset );
   }
+
+  emit verticalGridChanged();
```

Comment:
Random discovery when reviewing this: we currently create a proj object for each call to this, should we rather cache this when the vertical grid is set? https://github.com/opengisch/QField/pull/5917/files?diff=unified&w=1#diff-d7c2354458a68177df2d0253ffddfc748ca33192354477b781dd176e04f1fc37L152-R170

Reply:
@m-kuhn , interesting, I'm game to do this in a separate PR to reduce the noise here. I also spotted a similar optimization with the tracker I want to do in a separate PR.